estate/models/estate_property.py

```diff
@@ -0,0 +1,114 @@
+"""Odoo modules"""
+from dateutil.relativedelta import relativedelta
+from odoo import api, fields, models, exceptions
+from odoo.exceptions import ValidationError
+from odoo.tools import float_utils as fu
+
+class EstateProperty(models.Model):
+    '''Estate property'''
+    _name = "estate.property"
+    _description = "Property of an estate"
+    _order = "sequence, id desc"
+
+    name = fields.Char(required = True, string="Title")
+    sequence = fields.Integer('Sequence', default=1, help="Used to order stages. Lower is better.")
+    description = fields.Text("Description")
+    postcode = fields.Char("Postcode")
+    date_availability = fields.Date(
+        "Available From",
+        copy = False,
+        default = lambda self: self._default_date_availability()
+    )
+    expected_price = fields.Float("Expected Price", required = True)
+    selling_price = fields.Float("Selling Price", readonly = True, copy = False)
+    bedrooms = fields.Integer("Bedrooms", default = 2)
+    living_area = fields.Integer("Living Area (sqm)")
+    facades = fields.Integer("Facades")
+    garage = fields.Boolean("Garage")
+    garden = fields.Boolean("Garden")
+    garden_area = fields.Integer("Garden Area (sqm)")
+    garden_orientation = fields.Selection(
+        string = "Orientation",
+        selection = [('north', 'North'), ('east', 'East'), ('south', 'South'), ('west', 'West')]
+    )
+    active = fields.Boolean("Active", default = True)
+    state = fields.Selection(
+        string = "Status",
+        selection = [('new', 'New'), ('offer_received', 'Offer Received'), ('offer_accepted', 'Offer Accepted'), ('sold', 'Sold'), ('cancelled', 'Cancelled')],
+        default = 'new',
+        required = True,
+        copy = False
+    )
+    property_type_id = fields.Many2one("estate.property.type", string = "Property Type")
+    buyer_id = fields.Many2one("res.partner", copy = False, string = "Buyer")
+    salesperson_id = fields.Many2one("res.users", default = lambda self: self.env.user, string = "Salesperson")
+    tag_ids = fields.Many2many("estate.property.tag", string = "Tags")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string = "Offers")
+    total_area = fields.Float(compute = "_compute_area", string = "Total Area")
+    best_price = fields.Float(compute = "_compute_best_price", string = "Best Offer")
+
+
+
+    _sql_constraints = [
+        ('check_selling_price', 'CHECK(selling_price >= 0)', 'The selling price must be positive.'),
+        ('check_expected_price', 'CHECK(expected_price > 0)', 'The expected price must be strictly positive.')
+    ]
+
+
+
+    @api.depends("living_area", "garden_area")
+    def _compute_area(self):
+        for record in self:
```

Comment:
we try to be more explicit. record is too generic.
Here we are manipulating properties, so name it : for property in self.
apply this logic everywhere